linux: initramfs: use initramfs config file(s) and directory structure

dhewg opened the pull request:
This was a PR created by @mglae at my fork (for whatever reason).
Cleaned up and rebased .

MilhouseVH commented:
Do we still need `cpio:host`?

The two configs:
```
packages/linux/initramfs.conf 
projects/Generic/linux/initramfs.x86_64.conf
```

I'm wondering if they should be associated with the `initramfs` package, rather than `linux`?

They should also be in a `config` sub-dir, ie.
```
packages/initramfs/config/initramfs.conf 
projects/Generic/packages/initramfs/config/initramfs.x86_64.conf
```
(Yes, the location of the linux kernel configs is a historical anachronism)

dhewg commented:
Indeed, it looks like we don't need the cpio dependency anymore, linux's usr/gen_init_cpio.c creates the archive itself. Removed it.
Also moved the config files to `initramfs/config`

MilhouseVH commented:
@dhewg thanks for updating the PR, however you've created:
 ```
projects/Generic/initramfs/config/initramfs.x86_64.conf
```
when this should be:
```
projects/Generic/packages/initramfs/config/initramfs.x86_64.conf
```
in order to maintain the correct package hierarchical structure.

I'd rather we don't create more package-specific subdirs in the root of each project (or device) - `kodi` and `linux` are pretty much the only two that do this, for legacy reasons (and we could/should tidy them up at some point, certainly `kodi`).

MilhouseVH commented:
Tested with Generic and RPi2 builds, all looks good to me - thanks!

MilhouseVH commented:
@hiassoft can you rebase this ('cpio' was bumped in #4210)

HiassofT commented:
@MilhouseVH no, that's not my PR :-)

MilhouseVH commented:
Doh, sorry! Ping @dhewg :smile:

dhewg commented:
Done ;)
